jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/utils/JMSUtils.java

```diff
@@ -0,0 +1,193 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.transport.jms.utils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.messaging.CarbonMessage;
+import org.wso2.carbon.messaging.MapCarbonMessage;
+import org.wso2.carbon.messaging.SerializableCarbonMessage;
+import org.wso2.carbon.messaging.TextCarbonMessage;
+import org.wso2.carbon.transport.jms.exception.JMSConnectorException;
+
+import java.util.Enumeration;
+import java.util.Properties;
+
+import javax.jms.BytesMessage;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.ObjectMessage;
+import javax.jms.TextMessage;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NameNotFoundException;
+import javax.naming.NamingException;
+import javax.naming.Reference;
+
+/**
+ * This class is maintains the common methods used by JMS transport.
+ */
+public class JMSUtils {
+
+    private static final Log log = LogFactory.getLog(JMSUtils.class);
+
+    /**
+     * Return the JMS destination with the given destination name looked up from the context.
+     *
+     * @param context         the Context to lookup
+     * @param destinationName name of the destination to be looked up
+     * @param destinationType type of the destination to be looked up
+     * @return the JMS destination, or null if it does not exist
+     */
+    public static Destination lookupDestination(Context context, String destinationName, String destinationType)
+            throws NamingException {
+        if (null == destinationName) {
+            return null;
+        }
+
+        try {
+            return JMSUtils.lookup(context, Destination.class, destinationName);
+        } catch (NameNotFoundException e) {
+            try {
+                Properties initialContextProperties = new Properties();
+                if (null != context.getEnvironment()) {
```

Comment:
are these null checks needed? Counter question would be what should be done if they are null? If we do not wish to continue we should let the exception to be thrown.